operators/src/processing/expression/parser.rs

```diff
@@ -407,73 +386,104 @@ mod tests {
         ast.into_token_stream().to_string()
     }
 
+    fn prelude() -> impl ToTokens {
+        quote! {
+            #[inline]
+            fn apply(a: Option<f64>, b: Option<f64>, f: fn(f64, f64) -> f64) -> Option<f64> {
+                match (a, b) {
+                    (Some(a), Some(b)) => Some(f(a, b)),
+                    _ => None,
+                }
+            }
+        }
+    }
+
     #[test]
     fn simple() {
+        let prelude = prelude();
+
         assert_eq!(
-            parse("expression", &[], &[], "1"),
+            parse("expression", &[], "1"),
             quote! {
+                #prelude
+
                 #[no_mangle]
-                pub extern "C" fn expression() -> f64 {
-                    1f64
+                pub extern "Rust" fn expression() -> Option<f64> {
```

Comment:
does it have any implications that we are generating rust now?

Reply:
It uses the Rust ABI and not the C ABI anymore

Reply:
yeah, but I meant: does it e.g. increase the compilation time significantly?

Reply:
No, we should just be careful testing this on our server, because I'm not sure how stable the Rust ABI is compared to the C ABI. With the C-ABI, we could ship pre-compiled versions to any system.